### PR TITLE
[Feature] Job Submission - Refreshed Baseline Implementation

### PR DESCRIPTION
**Pull request by s7monk**
### Purpose

Job submit baseline implementation.
The baseline implementation for job submission has been completed, with the Flink portion fully implemented. The standards for Spark's implementation have been defined and are ready for further development.

### CODE REVIEW

File: paimon-web-common/pom.xml [hunk 1]
```diff
@@ -38,11 +38,6 @@ under the License.
     </properties>
 
     <dependencies>
-        <dependency>
```
Comment (SteNicholas) [inline paimon-web-common/pom.xml:41]: Why remove this dependency? Is this change related with job submission?

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.common.executor;
+
+import org.apache.paimon.web.common.result.FetchResultParams;
+import org.apache.paimon.web.common.result.SubmitResult;
+
+/** The Executor interface. */
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java:24]: Could you describe this interface more detailed?

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public interface Executor {
+
+    SubmitResult executeSql(String statement) throws Exception;
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java:27]: Add the comment of all interface methods.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    SubmitResult fetchResults(FetchResultParams params) throws Exception;
+
+    boolean stop(String jobId, boolean withSavepoint, boolean withDrain) throws Exception;
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java:31]: Is the executor only defined for Flink job? What's the behavior of Spark job for `withSavepoint` and `withDrain` param?

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/ExecutorFactory.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.common.executor;
+
+/** The ExecutorFactory interface provides a method to create an Executor. */
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/ExecutorFactory.java:21]:
```suggestion
/** This factory is defined to create an {@link Executor} instance. */
```

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/ExecutorFactory.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+public interface ExecutorFactory {
+
+    Executor createExecutor() throws Exception;
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/ExecutorFactory.java:24]: Add the comment of all interface methods.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/function/Supplier.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.common.function;
+
+/** Supplier Functional interface that can use exceptions. */
+@FunctionalInterface
+public interface Supplier<T> {
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/function/Supplier.java:23]: Why introduce this interface? Why not use `java.util.function.Supplier`?

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/FetchResultParams.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+package org.apache.paimon.web.common.result;
+
+/** Represents the parameters required to fetch the results of a certain operation. */
+public class FetchResultParams {
+
+    private final String sessionId;
+    private final String submitId;
+    private final Long token;
+
+    private FetchResultParams(Builder builder) {
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/result/FetchResultParams.java:28]: I don't think the constructor need to use builder as parameter. The builder mode isn't used via this way.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+ */
+
+package org.apache.paimon.web.common.result;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** This class represents the result of a job. Including flink job and spark Job. */
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:26]: Why `SubmitRsult` represents the result of a job? IMO, the `SubmitRsult` only means the submission result of job. BTW, the `Including flink job and spark Job.` could be removed.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+public class SubmitResult implements Serializable {
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:27]:
```suggestion
public class SubmissionResult implements Serializable {
```

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+    private static final long serialVersionUID = 1L;
+
+    private final String submitId;
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:31]: What's the purpose of definition for `submitId`?

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        public SubmitResult build() {
+            return new SubmitResult(this);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "SubmitResult{"
```
Comment (SteNicholas) [inline paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:115]: Does the `shouldFetchResult` need to be included in `toString`?

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/exception/SqlExecutionException.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.flink.exception;
+
+/** Exception class for SQL execution issues. */
```
Comment (SteNicholas) [inline paimon-web-flink/src/main/java/org/apache/paimon/web/flink/exception/SqlExecutionException.java:21]:
```suggestion
/** Exception of SQL execution. */
```

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutorFactory.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+    public Executor createExecutor() {
+        EnvironmentSettings environmentSettings = getEnvironmentSettings(mode);
+        StreamExecutionEnvironment env =
+                StreamExecutionEnvironment.getExecutionEnvironment(configuration);
+        TableEnvironment tableEvn = StreamTableEnvironmentImpl.create(env, environmentSettings);
+        return new FlinkExecutor(env, tableEvn);
+    }
+
+    private EnvironmentSettings getEnvironmentSettings(RuntimeExecutionMode mode) {
+        return mode == RuntimeExecutionMode.BATCH
```
Comment (SteNicholas) [inline paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutorFactory.java:53]: What's the behavior of `RuntimeExecutionMode.DYNAMIC`?

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutorFactory.java [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+        this.mode = mode;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Executor createExecutor() {
+        EnvironmentSettings environmentSettings = getEnvironmentSettings(mode);
+        StreamExecutionEnvironment env =
+                StreamExecutionEnvironment.getExecutionEnvironment(configuration);
+        TableEnvironment tableEvn = StreamTableEnvironmentImpl.create(env, environmentSettings);
```
Comment (SteNicholas) [inline paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutorFactory.java:48]:
```suggestion
        TableEnvironment tableEnv = StreamTableEnvironmentImpl.create(env, environmentSettings);
```

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+                    } else if (insertStatements.isEmpty()) {
+                        return executeDmlStatement(statement);
+                    }
+                    break;
+                default:
+                    executeStatementWithoutResult(statement);
+                    break;
+            }
+        }
+        return executeInsertStatements(insertStatements);
```
Comment (SteNicholas) [inline paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java:77]: Is insert statement same as DML?

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Override
+    public SubmitResult executeSql(String multiStatement) throws SqlExecutionException {
+        List<String> insertStatements = new ArrayList<>();
+        String[] statements = StatementParser.parse(multiStatement);
+        for (String statement : statements) {
+            FlinkSqlOperationType operationType = FlinkSqlOperationType.getOperationType(statement);
+            switch (operationType.getCategory()) {
+                case DQL:
+                    if (insertStatements.isEmpty()) {
```
Comment (SteNicholas) [inline paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java:61]: Could empty insert statement be handled early?

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@s7monk, thanks for the contribution. This implementation pull request is huge and needs too much time to review. I left some comments for part of this pull request. PTAL.
BTW, the implementation of `Executor` does not support `fetchResult` and `stop`. Therefore, IMO, this methods could be removed in this pull request.

File: paimon-web-common/pom.xml (hunk 1 above)
Comment (s7monk) [reply paimon-web-common/pom.xml:41]: Thank you for bringing this to my attention. You're correct; this part isn't related to the current PR. I will take the revert it.

Comment (s7monk) [reply paimon-web-common/pom.xml:41]: Thank you for bringing this to my attention. You're correct; this part isn't related to the current PR. I will take the revert it.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java (hunk 2 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java:24]: ok.

File: paimon-web-common/pom.xml (hunk 1 above)
Comment (s7monk) [reply paimon-web-common/pom.xml:41]: Thank you for bringing this to my attention. You're correct; this part isn't related to the current PR. I will take the revert it.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java (hunk 4 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java:31]: Spark's stop method may not need withSavepoint and withDrain, I will pass a variable parameter.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/function/Supplier.java (hunk 7 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/function/Supplier.java:23]: OK, i will remove this.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/FetchResultParams.java (hunk 8 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/result/FetchResultParams.java:28]: Yes, I will modify this class.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java (hunk 9 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:26]: ok.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java (hunk 12 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:115]: I will remove.

File: paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java (hunk 11 above)
Comment (s7monk) [reply paimon-web-common/src/main/java/org/apache/paimon/web/common/result/SubmitResult.java:31]: If it is sql gateway, submitId will store operationHandle in order to obtain the results, and others will store a uuid.

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutorFactory.java (hunk 14 above)
Comment (s7monk) [reply paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutorFactory.java:53]: I ignored this, I will modify it.

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java (hunk 17 above)
Comment (s7monk) [reply paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java:61]: The statements passed by the sql editor can only contain one of DQL and Insert. If there is an insert statement in front of it, the DQL statement cannot be executed, so it is judged here first.

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java (hunk 16 above)
Comment (s7monk) [reply paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java:77]: If it contains multiple insert statements, statementSet will be opened, so insert is executed last.

Comment (SteNicholas) [reply paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java:77]: @s7monk, IMO, this could merge into DML branch, no need to distiguish here.

File: paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+import org.apache.flink.table.api.TableResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** The flink implementation of the {@link Executor}. */
+public class FlinkExecutor implements Executor {
+
+    private static final String EXECUTE_SUCCESS = "OK";
```
Comment (SteNicholas) [inline paimon-web-flink/src/main/java/org/apache/paimon/web/flink/executor/FlinkExecutor.java:44]: Where is the execution failure status?

**Review by SteNicholas — CHANGES_REQUESTED** [review]
